CPP-Example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   gets(src); 
```

Comment:
[Banned C function detected (gets)](https://github.com/microsoft/devskim/guidance/DS181021.md)
gets will read in as many bytes as are provided, regardless of the size of buffer recieving the bytes.  This allows for conditions that cause a buffer overlow in the destination buffer.Replacement: Use gets_s or fgets if possible. fgets has different behavior from gets (may include newline for example), so care should be taken if it is used.


Suggestions:
**[1]** *+   gets_s(src, <size of src>); *
**[2]** *+   fgets(src, <size of src>, stdin); *